src/backend/distributed/metadata/node_metadata.c

```diff
@@ -579,8 +579,9 @@ SetUpDistributedTableDependencies(WorkerNode *newWorkerNode)
 		 */
 		if (ClusterHasDistributedFunctionWithDistArgument())
 		{
-			MarkNodeHasMetadata(newWorkerNode->workerName, newWorkerNode->workerPort,
-								true);
+			bool localOnly = false;
```

Comment:
Why do we set `Anum_pg_dist_node_hasmetadata` when we only decide to trigger metadata sync?

`SetWorkerColumn` should only be done when the metadata syncing actually happens. We already do that in `SyncMetadataToNodes`.

I wonder if the hack introduced in `SetWorkerColumn` is needed due this?